bao_supplier_certificate/models/supplier_certificate_generator.py

```diff
@@ -0,0 +1,127 @@
+            ("confirmed", _("Confirmed")),
+            ("cancelled", _("Cancelled")),
+        ],
+        default="draft",
+    )
+    sent = fields.Boolean(
+        compute="_compute_sent",
+        help="Are all certificates sent ?",
+    )
+    supplier_ids = fields.Many2many(
+        string="Vendors",
+        comodel_name="res.partner",
+        help="Leave this field empty to automatically select Vendors "
+        "that have a Purchase Order confirmed between Start Date and "
+        "End Date.",
+    )
+    supplier_certificate_ids = fields.One2many(
+        string="Certificates",
+        comodel_name="bao.supplier.certificate",
+        inverse_name="supplier_certificate_generator_id",
+    )
+    supplier_certificate_count = fields.Integer(
+        compute="_compute_supplier_certificate_count",
+        string="Certificate Count",
+        copy=False,
+        default=0,
+        store=True,
+    )
+
+    @api.depends("supplier_certificate_ids")
+    def _compute_supplier_certificate_count(self):
+        """Compute the number of supplier_certificate_ids"""
+        for certificate_generator in self:
+            certificate_generator.supplier_certificate_count = len(
+                certificate_generator.supplier_certificate_ids
+            )
+
+    @api.depends("supplier_certificate_ids")
+    def _compute_sent(self):
+        """Compute the number of supplier_certificate_ids"""
+        for certificate_generator in self:
+            sent_certificates = certificate_generator.supplier_certificate_ids.filtered(
+                lambda r: r.state == "sent"
+            )
+            certificate_generator.sent = len(sent_certificates) == len(
+                certificate_generator.supplier_certificate_ids
+            )
+
+    def _get_related_supplier_certificate_domain(self):
+        """Return domain used to show related bao.supplier.certificate."""
+        return [("supplier_certificate_generator_id", "=", self.id)]
+
+    def action_supplier_certificate(self):
+        """This function returns an action that display existing
+        Supplier Certificate of the current Supplier Certificate
+        Generator.
+        """
+        action = self.env["ir.actions.act_window"]._for_xml_id(
+            "bao_supplier_certificate.action_supplier_certificate"
+        )
+        action["domain"] = self._get_related_supplier_certificate_domain()
+        return action
+
+    def action_confirm(self):
+        """Confirme Certificate Generator and generate Certificates"""
+        for generator in self:
+            # Fill in supplier_ids if empty
+            if not generator.supplier_ids:
+                purchase_order_ids = self.env["purchase.order"].search(
+                    [
+                        ("state", "in", ["purchase", "done"]),
+                        ("date_approve", ">=", self.start_date),
+                        ("date_approve", "<=", self.end_date),
+                    ]
+                )
```

Comment:
Extract this to a separate method, probably. It also has _a lot_ of overlap with `_compute_purchase_order_ids` in `bao.supplier.certificate`, so we can probably reduce the repeated code somehow.

Reply:
Added to Roadmap.